flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/DefaultJobGraphStore.java

```diff
@@ -241,6 +243,22 @@ public void putJobGraph(JobGraph jobGraph) throws Exception {
         LOG.info("Added {} to {}.", jobGraph, jobGraphStateHandleStore);
     }
 
+    @Override
+    public void putJobResourceRequirements(
+            JobID jobId, JobResourceRequirements jobResourceRequirements) throws Exception {
+        synchronized (lock) {
+            @Nullable final JobGraph jobGraph = recoverJobGraph(jobId);
+            if (jobGraph == null) {
+                throw new FileNotFoundException(
```

Comment:
I think `java.util.NoSuchElementException` would be more appropriate here because it doesn't necessarily involves files.